core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java

```diff
@@ -525,6 +525,51 @@ boolean isLengthInvalidOrZero() {
     return this.length < 1;
   }
 
+  /**
+   *
+   * @return get length in bytes (dealing with special characters)
+   */
+  int getLengthInBytes( String value ) {
+    int total = 0;
+    try {
+      for ( int i = 0; i < value.length(); i++ ) {
+        String c = new String( new char[] {value.charAt( i )} );
+        int length = c.getBytes( "UTF-8" ).length;
+        total += length;
+      }
+      return total;
+
+    } catch ( UnsupportedEncodingException e ) {
+      return value.length();
+    }
+  }
+
+  /**
+   *
+   * @return get length in bytes (if const is defined or database only supports char validation ) or chars (default
+   * except for Oracle Databases)
+   */
+
+  int getLengthInCharOrBytes( String string, DatabaseMeta databaseMeta )  {
+    String val = EnvUtil.getSystemProperty( Const.KETTLE_COMPATIBILITY_LENGTH_VALIDATION_USES_BYTES );
```

Comment:
In this method we just call `lengthSemanticsUsesChars() ` on the given `databaseMeta` and decide if we just return the `string.length()` or the length in bytes. A single ternary operator will do.